core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +134,19 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
+    public void validateStackStatusForStartHostGroup(StackDto stack) {
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
+        List<String> unhealthyHostsGroupNames = processor.getUnhealthyDependentHostGroups(stack);
+        if (CMRepositoryVersionUtil.isCmServicesHealthCheckAllowed(runtimeVersionService.getRuntimeVersion(stack.getCluster().getId()))) {
+            if (stack.getStack().isModificationInProgress()) {
+                throw new BadRequestException(format("Data Hub '%s' has '%s' state. Upscaling is not allowed.",
+                        stack.getStack().getName(), stack.getStack().getStatus()));
+            } else if (!unhealthyHostsGroupNames.isEmpty()) {
+                throw new BadRequestException(format("Upscaling is Not Allowed as Data hub '%s' has Unhealthy Nodes", stack.getStack().getName()));
```

Comment:
Please include information about which hostGroup is unhealthy in the exception, as well as which hostGroup we were trying to scale.